source/slang/slang-serialize-container.h

```diff
@@ -0,0 +1,107 @@
+// slang-serialize-container.h
+#ifndef SLANG_SERIALIZE_CONTAINER_H
+#define SLANG_SERIALIZE_CONTAINER_H
+
+#include "../core/slang-riff.h"
+#include "slang-serialize-types.h"
+#include "slang-ir-insts.h"
+#include "slang-profile.h"
+
+namespace Slang {
+
+class EndToEndCompileRequest;
+
+/* The binary representation actually held in riff/file format*/
+struct SerialContainerBinary
+{
+    struct Target
+    {
+        uint32_t target;
+        uint32_t flags;
+        uint32_t profile;
+        uint32_t floatingPointMode;
+    };
+
+    struct EntryPoint
+    {
+        uint32_t name;
+        uint32_t profile;
+        uint32_t mangledName;
+    };
+};
+
+/* Struct that holds all the data that can be held in a 'container' */
+struct SerialContainerData
+{
+    struct Target
+    {
+        CodeGenTarget       codeGenTarget = CodeGenTarget::Unknown;
+        SlangTargetFlags    flags = 0;
+        Profile             profile;
+        FloatingPointMode   floatingPointMode = FloatingPointMode::Default;
+    };
+    
+    struct TargetModule
+    {
```

Comment:
The equivalent type in `slang-compiler.h` is currently called `TargetProgram`, and in reality it acts more like a `TargetComponentType` or `TargetSpecializedComponentType`.

The name `TargetModule` is a bit misleading here, because while this thing does hold an `IRModule`, it doesn't relate to the notion of a "module" as understood by the non-IR parts of the compiler.

It is probably not necessary to come up with a different name right now, but we will eventually need a consistency pass where we try to get our terminology the same across as many parts of the compiler as possible.

Reply:
I'm happy to rename - as with some of these other types, it wasn't really clear what was appropriate.